core/lib/dal/src/sync_dal.rs

```diff
@@ -64,40 +63,54 @@ impl SyncDal<'_, '_> {
         .instrument("sync_dal_sync_block.block")
         .with_arg("block_number", &block_number)
         .fetch_optional(self.storage.conn())
+        .await?
+        else {
+            return Ok(None);
+        };
+        Ok(Some(block.try_into()?))
+    }
+
+    pub(super) async fn sync_block_transactions(
+        &mut self,
+        block_number: MiniblockNumber,
+    ) -> sqlx::Result<Vec<Transaction>> {
+        let transactions = sqlx::query_as!(
+            StorageTransaction,
+            r#"
+            SELECT
+                *
+            FROM
+                transactions
+            WHERE
+                miniblock_number = $1
+            ORDER BY
+                index_in_block
+            "#,
+            block_number.0 as i64
+        )
+        .instrument("sync_dal_sync_block.transactions")
+        .with_arg("block_number", &block_number)
+        .fetch_all(self.storage.conn())
         .await?;
+        Ok(transactions.into_iter().map(Transaction::from).collect())
+    }
 
-        let Some(storage_block_details) = storage_block_details else {
+    pub async fn sync_block(
+        &mut self,
+        block_number: MiniblockNumber,
+        current_operator_address: Address,
+        include_transactions: bool,
```

Comment:
ideally whether transactions are included should be reflected on the type level. Do you think there is a way to do this?

Reply:
flag indicating whether or not transactions are requested is an argument of the rpc call: https://github.com/matter-labs/zksync-era/blob/39f2d5057a7bb5b489e858bafb72fb925ebc6c8f/core/lib/zksync_core/src/api_server/web3/namespaces/en.rs#L19 , so at best we could make this method always include transactions. Alternatively we can also
* make the the rpc call handler call 2 functions (one for header, one for transactions) instead, so that there is no boolean flag in dal.
* Create a newtype for this argument `struct IncludeTransactions(boolean)`, to make function calls more readable.